### PR TITLE
Switch CI to Github Actions

### LINKED ISSUE #118: Add Github Actions CI Workflow
Need to add a CI. Should test multiple ruby versions. Will create a PR for this, I have a good template.

### PR DESCRIPTION
**Pull request by westonganger**
Resolves #118 

Notable changes:

**jruby-1.9mode** - removed from test suite because setup-ruby doesnt support it
**jruby-9.3** - was added
**ruby-head, jruby-head** - removed because it offers us no value, also because continue-on-error feature is not working here, passes in actions view, but shows as failing on PRs for some reason

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: Install ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: "${{ matrix.ruby }}"
+
+    - name: Bundle
+      run: |
+        gem install bundler
+        bundle install
```
Comment (aried3r) [inline .github/workflows/test.yml:42]: If we added `bundler-cache: true` to the `ruby/setup-ruby` task, we could remove the Bundle step altogether.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - ruby: 2.3
+          - ruby: 2.4
+          - ruby: 2.5
+          - ruby: 2.6
+          - ruby: 2.7
+          - ruby: 3.0
```
Comment (aried3r) [inline .github/workflows/test.yml:26]: Could you also add Ruby 3.1?

File: .github/workflows/test.yml (hunk 1 above)
Comment (simi) [reply .github/workflows/test.yml:42]: :+1:

File: .github/workflows/test.yml (hunk 2 above)
Comment (simi) [reply .github/workflows/test.yml:26]: :+1:

**Comment by simi** [comment]
Sorry, this felt out of my radar. Can you address PR comments @westonganger? Thanks @aried3r for nice review :heart:.

File: .github/workflows/test.yml (hunk 2 above)
Comment (westonganger) [reply .github/workflows/test.yml]: Added.

File: .github/workflows/test.yml (hunk 1 above)
Comment (westonganger) [reply .github/workflows/test.yml]: Updated

File: .github/workflows/test.yml (hunk 2 above)
Comment (simi) [reply .github/workflows/test.yml]: @westonganger Would you mind to update timecop in gemspec to `~> 0.9.0`? That way it should work on Ruby 3.1 as well.

Comment (westonganger) [reply .github/workflows/test.yml]: Ok updated.

**Comment by westonganger** [comment]
Now we have a failure on jruby 9.1 but not on jruby 9.2, can we drop 9.1 from the test suite?

**Comment by aried3r** [comment]
I'm not a maintainer, so I cannot speak on behalf of them, but I'd say we drop JRuby 9.1. It targets Ruby 2.3 and it seems that MRI 2.3 works fine, so on first glance it doesn't appear to be a "real" bug, as in, the fault is with caxlsx. Although I can't be sure.

This looks very good so far!

**Comment by simi** [comment]
Yup, feel free to drop it.

**Comment by westonganger** [comment]
Ok dropped jruby 9.1, the test suite is now fully passing.

**Comment by simi** [comment]
Briliant! :gem:

**Pull request merged**